infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
There's no spec at all, just that the core Java library happens to specify how hashCodes are implemented, and by luck that hasn't changed... Yes, it probably is perilous to continue using hashCodes. I'm converging more and more that we should just dump this approach and figure out how to tell the serializer to generate our hash code for us.